### ISSUE TITLE
Low power mode on iOS

### ISSUE THREAD

**Issue by smilindave26** [openziti/ziti-tunnel-apple#105]
E.g., Something like this if done outside of direct support in TSDK

* on phone sleep, disconnect Ziti (disabling identities will be insufficient - the Ziti connection needs to be shutdown to prevent attempts to access the controller and ERs)
* on wake, reconnect
* Button on UI to enable/disable this mode (presuming since some apps will need connectivity while phone is sleeping

Alt: consider supporting this mode in TSDK (e.g., connect to ERs only on Dial attempt - perceived performance would be impacted, and doesn't address hosting services...)

**Comment by smilindave26** [comment]
Also - make sure that "low power mode" requires interceptDnsByMatchingDomains==true to avoid DNS traffic routing through the appex...

**Comment by smilindave26** [comment]
some notes from The Eskimo: https://developer.apple.com/forums/thread/95988

**Comment by smilindave26** [comment]
Disabling identities on sleep and re-enabling on wake so fair if a big failure.  When not in Do-no-disturb, phone sleeps and wakes up very vert frequently, having the opposite of intended effect on the battery.

Need a new strategy:
a) only disable if been asleep for some time period (e.g., a couple of mins)?
b) only disable if also in Do-not-disturb mode?
c) add a setting in TSDK or CSDK that allows a quick "tune down" (dramatically slow down things like service requests and posture checks, anything else running on a timer...)?
d) Just setting tunnel networking to `nil` so no apps using the tunnel can cause traffic?  prob not enough.  maybe combine with c)?

**Comment by mvelbaum** [comment]
I was wondering if this can be prioritized again? As of right now the mobile client massively drains the battery, making it impractical to leave it enabled all the time. It would be great if it could get to the level of the WireGuard iOS client, which consumes very little power.

**Comment by smilindave26** [comment]
Yes. We have heard from a handful of users that they are having this issue and are discussing potential approaches to address it.  When we have something, I'll post it in [Discourse](https://openziti.discourse.group/) and on this issue and solicit users to join a TestFlight.
